server/get_payload.go

```diff
@@ -248,7 +262,7 @@ func (m *BoostService) innerGetPayload(log *logrus.Entry, signedBlindedBeaconBlo
 			var result payloadResult
 			result.success = true
 
-			if version == GetPayloadV1 {
+			if versionToUse == GetPayloadV1 {
```

Comment:
We should definitely be validating the returned block by the relay even if we fall back to the v1 API.